### ISSUE TITLE
The constraint 0 < d+1 does not seem to resolve?

### ISSUE THREAD

**Issue by noinia** [clash-lang/ghc-typelits-natnormalise#70]
In the following little code snippet I have to functions, function "libFunc", which has a constraint i < d for two Typelits (e.g. indexing with an index i into some sort of vector of length d).  If I use libFunc to index element 0 in a vector of length (d+1) I get the constraint 0 < d+1. This constraint is vacuously true, and hence I was expecting the plugin to resolve this constraint for me. Is this a bug, or was I expecting too much?

I'm using ghc-9.2.4 and ghc-typelits-natnormalise-0.7.7.

```haskell
{-# LANGUAGE AllowAmbiguousTypes#-}
{-# LANGUAGE DataKinds #-}
{-# LANGUAGE GADTs #-}
{-# OPTIONS_GHC -fplugin GHC.TypeLits.KnownNat.Solver #-}
{-# OPTIONS_GHC -fplugin GHC.TypeLits.Normalise #-}
module Bug where

import GHC.TypeLits
import Data.Type.Ord

libFunc :: forall (i :: Nat) d. i < d => ()
libFunc = ()

-- useFunc :: forall (d :: Nat). 0 < d+1 => () -- with this signature it works
useFunc :: forall (d :: Nat). ()
useFunc = libFunc @0 @(d+1)
```

**Comment by rowanG077** [comment]
GHC 9.2 changed a lot of things about how type level comparisons are done. The gist of it is that prior to 9.2 we only had to do reasoning involving `<=` now we have the entire set of possible comparisons.

While the plugin was updated to support GHC 9.2 and latter the reasoning it does still only works on `<=` and not yet on the other inequalites.

See here: https://github.com/clash-lang/ghc-typelits-natnormalise/blob/master/src-pre-ghc-9.4/GHC/TypeLits/Normalise.hs#L758

for the code fragment that only matches on `OrdCond (Compare m n) 'True 'True 'False` which is the type alias of  `<=?`.

To fix(untested but should work) for your code is to rewrite it in terms of `<=`:

```
libFunc :: forall (i :: Nat) d. i <= (d - 1) => ()
libFunc = ()

useFunc :: forall (d :: Nat). ()
useFunc = libFunc @0 @(d+1)
```

**Comment by noinia** [comment]
Hmm, changing libFunc to have the constraint i <= d-1 indeed allows me to circumvent the issue. 

How much work do you expect it to be to include support for the other comparison types? I.e. would that be something that can be fixed relatively "locally" to the lines 751-771 there? E.g. would it be possible to deal with the x < y case by e.g. rewriting x' to whatever the appropriate representation of (1+x') is? 

If so I might take a stab at trying to fix that. However, the documentation around what is happening there is somewhat sparse.....

**Comment by rowanG077** [comment]
Yeah I'd expect you could at least get some rudimentary support for the other operators by just rewriting mechanically to `<=`. Probably refactor into a separate function  as that function is a bit cluttered as is.
